server/base/src/main/java/org/apache/accumulo/server/fs/FileManager.java

```diff
@@ -244,8 +244,9 @@ private List<String> takeOpenFiles(Collection<String> files,
     return filesToOpen;
   }
 
-  private Map<FileSKVIterator,String> reserveReaders(KeyExtent tablet, Collection<String> files,
-      boolean continueOnFailure, CacheProvider cacheProvider) throws IOException {
+  private Map<FileSKVIterator,TabletFile> reserveReaders(KeyExtent tablet,
+      Collection<TabletFile> files, boolean continueOnFailure, CacheProvider cacheProvider)
+      throws IOException {
 
     if (!tablet.isMeta() && files.size() >= maxOpen) {
```

Comment:
`files` may contain multiple instances that reference the same underlying file, just with a different range, right? This would seem to me that there is a higher probability that users will hit the limit set by `TSERV_SCAN_MAX_OPENFILES` and other file limit properties. I think we need to take this into account when checking the number of open files against their respective properties.

Reply:
I would think that in this case we could just track open files by the actual file and not use the ranges. It would make sense to me that you'd only care about real files that were open in HDFS so if you had several files that were the same but different ranges we would just count it as 1 and then this wouldn't be an issue.